api/server_helpers_test.go

```diff
@@ -337,6 +337,25 @@ func createExplorerServerTester(name string) (*serverTester, error) {
 	return st, nil
 }
 
+// decodeErrorResponse returns an error if the response's status code is
+// non-2xx. If an error message or Error is included in the response, it is
+// decoded and returned as the error.
+func decodeErrorResponse(resp *http.Response) error {
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
```

Comment:
this function is a little weird, since it may not actually decode anything at all. I'd prefer the conditional to be outside the function; something like:
```go
if non2xx(resp.StatusCode) {
    return decodeError(resp)
}
```